FS logging backend assumes littlefs

**Describe the bug**
FS Logging backend is assumed to be used with littlefs

**To Reproduce**
Steps to reproduce the behavior:
1. set: 
<pre>
CONFIG_FILE_SYSTEM=y
CONFIG_FAT_FILESYSTEM_ELM=y

CONFIG_LOG_BACKEND_FS=y
</pre>
2. Got build error: `cannot find -lLITTLEFS`

**Expected behavior**
FS Logging backend should be usable with other filesystems.

**Impact**
Cannot use FAT FS

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: espressif
 - Commit SHA or Version used: v2.6.0


**Additional context**
Also, the help text on KConfig for CONFIG_LOG_BACKEND_FS explicitly mentions "littlefs"

BTW, there's also this warning:
<pre>
/home/v01d/coding/embebit/ble_tracker/zephyr/subsys/logging/log_backend_fs.c:459:10: warning: initialization of 'void (*)(const struct log_backend * const)' from incompatible pointer type 'void (*)(void)' [-Wincompatible-pointer-types]
  .init = log_backend_fs_init,
</pre>

--- a/subsys/logging/CMakeLists.txt
+++ b/subsys/logging/CMakeLists.txt
@@ -26,8 +26,6 @@ if(NOT CONFIG_LOG_MINIMAL)
     log_backend_fs.c
   )
 
-  zephyr_link_libraries_ifdef(CONFIG_LOG_BACKEND_FS LITTLEFS)
-
   zephyr_sources_ifdef(
     CONFIG_LOG_CMDS
     log_cmds.c

--- a/subsys/logging/Kconfig.backends
+++ b/subsys/logging/Kconfig.backends
@@ -298,10 +298,13 @@ config LOG_BACKEND_ADSP
 	  family of audio processors
 
 config LOG_BACKEND_FS
-	bool "Enable LittleFS backend"
+	bool "Enable file system backend"
 	depends on FILE_SYSTEM
 	help
-	  When enabled backend is using LittleFS to output logs.
+	  When enabled, backend is using the configured file system to output logs.
+	  As the file system must be mounted for the logging to work, it must be
+	  either configured for auto-mount or manually mounted by the application.
+	  Log messages are discarded as long as the file system is not mounted.
 
 if LOG_BACKEND_FS
 

